src/plotting.jl

```diff
@@ -0,0 +1,28 @@
+
+interval_markers(x::AbstractInterval) = interval_markers(inclusivity(x))
+interval_markers(x::Inclusivity) = interval_markers(first(x), last(x))
+interval_markers(start::Bool, stop::Bool) = interval_markers.([start, stop])
+interval_markers(x::Bool) = x ? :vline : :none
+
+@recipe function f(xs::AbstractVector{<:AbstractInterval{T}}, ys) where T
+    new_xs = T[]
+    new_ys = []
+    markers = Symbol[]
+    for (x,y) in zip(xs, ys)
+        # To cause line to not be connected, need to add a breaker point with
+        # NaN in one of the coordinates. We pute that in `new_ys` as that is probably a
```

Comment:
hmm, is it safe to assume the NaN will be compatible for all `ys`?

Reply:
Not really, no.
But we can deal with that if and when it comes up.

It is compatible with the `new_ys` array.
But sometimes the recipes do weird things.

Reply:
I guess it's hard to predict what will break in advance.

Reply:
Unfortunately you can't both have the flexibility of having `x` accept arbitrary types, like Dates, and a guarantee that y is floats. Maybe you could have the `newys` constructor be `newys = Float64[]` so at least you'll get a reasonable error when trying to promote to floats. 
The only types that could be passed in as ys that would cause this recipe to fail, but not Plots itself, is anything that has a `@recipe f(::Type{MyType}, ::MyType)` signature defined for them and that cannot be promoted to floats. That's probably rare, but could happen. You could define that conversion yourself if you've got control of the type (you could even have an error message suggesting it I guess).